### PR TITLE
[Fastlane.swift] Fix method hanging and propose an unified lint method

### PR DESCRIPTION
**Pull request by minuscorp**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

As raised in #16677 the new version of Fastlane.swift hang when certain error clauses converged at the same time between the Ruby and the Swift part. This PR aims to address it.

Also, I've seen that the Swift codebase doesn't follow any lint rules nor common syntax rules, so the first step is to integrate Swiftformat to ensure that every API or external change gets linted properly into the codebase. This not makes conflict to propose changes or cleanup code as proposed by another members.

### CODE REVIEW

File: fastlane/lib/fastlane/swift_fastlane_function.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def initialize(action_name: nil, action_description: nil, action_details: nil, k
 
       # rubocop:disable LineLength
       # class instance?
-      @reserved_words = %w[associativity break case catch class continue convenience default deinit didSet do else enum extension fallthrough false final for func get guard if in infix init inout internal lazy let mutating nil operator override postfix precedence prefix private public repeat required return self set static struct subscript super switch throws true try var weak where while willSet].to_set
+      @reserved_words = %w[associativity break case catch class continue convenience default deinit didSet do else enum extension fallthrough false final for func guard if in infix init inout internal lazy let mutating nil operator override precedence private public repeat required return self static struct subscript super switch throws true try var weak where while willSet].to_set
```
Comment (janpio) [inline fastlane/lib/fastlane/swift_fastlane_function.rb:34]: What is this about?

File: fastlane/swift/ArgumentProcessor.swift [hunk 2]
```diff
@@ -19,68 +19,68 @@ struct ArgumentProcessor {
...
         let fastlaneArgs = stride(from: 1, to: args.count - 1, by: 2).map {
-            RunnerArgument(name: args[$0], value: args[$0+1])
+            RunnerArgument(name: args[$0], value: args[$0 + 1])
         }
         self.args = fastlaneArgs
-        
+
         let fastlaneArgsMinusLanes = fastlaneArgs.filter { arg in
-            return arg.name.lowercased() != "lane"
+            arg.name.lowercased() != "lane"
```
Comment (janpio) [inline fastlane/swift/ArgumentProcessor.swift:31]: no functional change here?
(I don't know any swift)

**Review by janpio — CHANGES_REQUESTED** [review]
Can we possibly separate the fix and the link thigns into two PRs?

I could not really find the method fix as there was so much unrelated formatting and code style going on.
(Alternatively a link to the location of the actual change would be nice as well.)

File: fastlane/lib/fastlane/swift_fastlane_function.rb (hunk 1 above)
Comment (minuscorp) [reply fastlane/lib/fastlane/swift_fastlane_function.rb:34]: This are not required reserved words in Swift 4.0. There was a time that they were in Swift 3 where you had to backtick them, but not in the current project version

File: fastlane/swift/ArgumentProcessor.swift (hunk 2 above)
Comment (minuscorp) [reply fastlane/swift/ArgumentProcessor.swift:31]: Let me double-check, but the return is unneeded.

File: fastlane/swift/SocketClient.swift [hunk 3]
```diff
@@ -27,121 +27,121 @@ public enum SocketClientResponse: Error {
...
+
+        socketStatus = .ready
+        socketDelegate?.connectionsOpened()
     }
-    
+
     public func send(rubyCommand: RubyCommandable) {
         verbose(message: "sending: \(rubyCommand.json)")
         send(string: rubyCommand.json)
+        writeSemaphore.signal()
```
Comment (minuscorp) [inline fastlane/swift/SocketClient.swift:120]: This is the line that was causing the hang

File: fastlane/swift/ArgumentProcessor.swift (hunk 2 above)
Comment (minuscorp) [reply fastlane/swift/ArgumentProcessor.swift:31]: Yeah, no `return` keyword is needed in one-line methods like `map`, `flatMap`, `filter`, etc.

File: fastlane/swift/SocketClient.swift (hunk 3 above)
Comment (minuscorp) [reply fastlane/swift/SocketClient.swift:120]: @janpio You're right I should've splitted the work in two tasks, but the fix was only one line, and the linting stuff was something it was in my head for a while now. I hope this serves. The explanation is more or less simple, there was a point were when the Ruby socket sent a failure and the Swift part tried both to write concurrently, causing a deadlock. This frees the write channel when that stack of functions is called.

Comment (janpio) [reply fastlane/swift/SocketClient.swift:120]: Wouldn't it be easier to quickly create a second PR for that 1 line change then? This PR will then be just about the linting after the other one is merged and this one rebased.

**Pull request closed without merging**